pkg/analysis/l2/l2analyzer.go

```diff
@@ -0,0 +1,96 @@
+package l2
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"nodemon/pkg/entities"
+	"strconv"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+type Response struct {
+	Jsonrpc string `json:"jsonrpc"`
+	ID      string `json:"id"`
+	Result  string `json:"result"`
+}
+
+func hexStringToInt(hexString string) (int64, error) {
+	// Parse the hexadecimal string to integer
+	return strconv.ParseInt(hexString[2:], 16, 64)
+}
+
+func collectL2Height(url string, ch chan<- int64, logger *zap.Logger) {
+	requestBody, err := json.Marshal(map[string]interface{}{
+		"jsonrpc": "2.0",
+		"id":      "1",
+		"method":  "eth_blockNumber",
+		"params":  []interface{}{},
+	})
+	if err != nil {
+		logger.Error("Failed to build a request body for l2 node", zap.Error(err))
+	}
+
+	resp, err := http.Post(url, "application/json", bytes.NewBuffer(requestBody))
+	if err != nil {
+		logger.Error("Failed to send a request to l2 node", zap.Error(err))
+		return
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
```

Comment:
If you know the limit, please let me know. It's not important here; there has to be a balance between code complexity and security. In the assumed scenario where more data is sent to the monitoring service, the attacker would have to compromise the L2 node or the network it's connected to. If one of these is compromised, instead of sending more data, the attacker could instead send false data, which would make more sense. But we don't validate the data that is sent to the monitoring service, we sort of trust it. If we don't validate the data that is sent, there's even less point to limit the amount of data received.

Reply:
Fair enough, agree